File: backend/maven/src/main/java/com/dreamteam/atelier/model/domain/Season.java
package com.dreamteam.atelier.model.domain;

import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.TypeAlias;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.Document;


import java.util.UUID;

/**
 * Базовый тип описывающий сущность - сезон
 */
@Document(collection = "seasons")
@TypeAlias(value = "season")
@Data
@EqualsAndHashCode
@RequiredArgsConstructor
public class Season {

    @Id
    private String id;
    @Indexed(unique = true)
    private UUID uuid;
    @Indexed(unique = true)
    private String name;
    private String description;

}

File: backend/nodejs/src/data/model/season.ts
import { Document, Schema, model } from 'mongoose';

export interface ISeason {
    name: string;
    description?: string;
}

export interface ISeasonDocument extends ISeason, Document {
}

export const SeasonSchema = new Schema({
    name: {
        type: String,
        required: true
    },
    description: {
        type: String
    }
});

export default model<ISeasonDocument>('Season', SeasonSchema);
